src/index.ts

```diff
@@ -0,0 +1,2719 @@
+      Decimal.dNegOne.add(Decimal.sub(term1, term2));
+    }
+  }
+
+  var slog_criticalfunction_3 = function(x, z) {
+    z = z.toNumber();
+    var lnx = x.ln();
+    var lnx2 = lnx.sqr();
+    var lnx3 = lnx.cube();
+    if (lnx.layer === 0 && lnx2.layer === 0 && lnx3.layer === 0)
+    {
+      lnx = lnx.toNumber();
+      lnx2 = lnx2.toNumber();
+      lnx3 = lnx3.toNumber();
+
+      var term1 = 6*z*(lnx+lnx3);
+      var term2 = 3*z*z*(3*lnx2-2*lnx3);
+      var term3 = 2*z*z*z*(1-lnx-2*lnx2+lnx3);
+      var top = term1+term2+term3;
+      var bottom = 2+4*lnx+5*lnx2+2*lnx3;
+
+      return -1 + top/bottom;
+    }
+    else
+    {
+      var term1 = (lnx.add(lnx3)).mul(6*z);
+      var term2 = (lnx2.mul(3).sub(lnx3.mul(2))).mul(3*z*z);
+      var term3 = (Decimal.dOne.sub(lnx).sub(lnx2.mul(2)).add(lnx3)).mul(2*z*z*z);
+      var top = term1.add(term2).add(term3);
+      var bottom = new Decimal(2).add(lnx.mul(4)).add(lnx2.mul(5)).add(lnx3.mul(2));
+
+      return Decimal.dNegOne.add(top.div(bottom));
+    }
+  }*/
+
+  //Function for adding/removing layers from a Decimal, even fractional layers (e.g. its slog10 representation).
+  //Everything continues to use the linear approximation ATM.
+  public layeradd10(diff: DecimalSource): Decimal {
+    diff = Decimal.fromValue_noAlloc(diff).toNumber();
+    const result = D(this);
+    if (diff >= 1) {
+      const layeradd = Math.trunc(diff);
+      diff -= layeradd;
+      result.layer += layeradd;
+    }
+    if (diff <= -1) {
+      const layeradd = Math.trunc(diff);
+      diff -= layeradd;
+      result.layer += layeradd;
+      if (result.layer < 0) {
+        for (let i = 0; i < 100; ++i) {
+          result.layer++;
+          result.mag = Math.log10(result.mag);
+          if (!isFinite(result.mag)) {
+            return result;
+          }
+          if (result.layer >= 0) {
+            break;
+          }
+        }
+      }
+    }
+
+    //layeradd10: like adding 'diff' to the number's slog(base) representation. Very similar to tetrate base 10 and iterated log base 10. Also equivalent to adding a fractional amount to the number's layer in its break_eternity.js representation.
+    if (diff > 0) {
+      let subtractlayerslater = 0;
+      //Ironically, this edge case would be unnecessary if we had 'negative layers'.
+      while (Number.isFinite(result.mag) && result.mag < 10) {
+        result.mag = Math.pow(10, result.mag);
+        ++subtractlayerslater;
+      }
+
+      //A^(10^B) === C, solve for B
+      //B === log10(logA(C))
+
+      if (result.mag > 1e10) {
+        result.mag = Math.log10(result.mag);
+        result.layer++;
+      }
+
+      //Note that every integer slog10 value, the formula changes, so if we're near such a number, we have to spend exactly enough layerdiff to hit it, and then use the new formula.
+      const diffToNextSlog = Math.log10(Math.log(1e10) / Math.log(result.mag), 10); // FIXME: This has an extra argument
```

Comment:
Broken code, extra argument
```suggestion
      const diffToNextSlog = Math.log10(Math.log(1e10) / Math.log(result.mag));
```